### ISSUE TITLE
Implementing custom RNN node

### ISSUE THREAD

**Issue by SourKream** [fchollet/keras#2240]
Hi, I am trying to implement a word by word attention model and one of steps i need implement is a custom RNN node. A SimpleRNN node implements the following equations,

> s_t = Tanh(U x_t + W s_{t-1})
> o_t = Softmax(V s_t)

where s_t is the state variable for the RNN and o_t is the output
I want to implement a custom RNN node which does the following,

> s_t = x_t + Tanh(W s_{t-1})
> o_t = s_t

Can this be done using Keras? Please suggest any possible methods.

I am building on the code provided here #2067. I think this is a related issue #2017

**Comment by matthiasplappert** [comment]
I'd suggest to just implement a custom layer yourself. You can use the Keras backend to make it work on Theano and TensorFlow. This should be fairly straightforward if you use an appropriate RNN (e.g. the SimpleRNN or LSTM) as a starting point and adapt it as needed.
